### DISCUSSION TITLE
deploy pulsar function manually? (kubernetes)

### DISCUSSION THREAD

**Discussion by zzzz465** [apache/pulsar#20787] [Q&A] [answered]
I'm using the Pulsar function but I feel uncomfortable using admin CLI to create the job.
so, as I found pulsar function in kubernetes is deployed as statefulset, I want to manually create statefulset.

is it possible?

**Comment by tisonkun** [comment]
cc @nlu90 @ericsyh do you have any input here?

**Reply by ericsyh** [reply, marked as answer]
@zzzz465 Function Mesh https://functionmesh.io can help on this use case b/c it uses a k8s operator to make Function, Connector as a CRD so you can manually create function job statefulset through the CRD API.
https://functionmesh.io/docs/functions/run-function/run-java-function
